lib/devices/set_system_date_and_time.ex

```diff
@@ -13,6 +13,13 @@ defmodule Onvif.Devices.SetSystemDateAndTime do
   end
 
   def request_body(%SystemDateAndTime{} = system_date_time) do
+      request_body([config: system_date_time, set_time?: false])
+  end
+
+  #def request_body(%SystemDateAndTime{} = system_date_time, opts) do
+  def request_body([config: %SystemDateAndTime{} = system_date_time] = opts) do
```

Comment:
Pattern matching on a keyword list is tricky because they're order-dependant. Doc reference: https://hexdocs.pm/elixir/1.14.5/Keyword.html#module-duplicate-keys-and-ordering.

Reply:
it was already solved on the last commit